src/parsers/helper.ts

```diff
@@ -52,11 +55,14 @@ export const plugins: ParserPlugin[] = [
 ];
 
 export const parseOptions = (filePath: string, strictMode = false): ParserOptions | null => {
-  const fileType = supportedFileType(filePath);
+  const fileType = getFileType(filePath);
   if (fileType === 'ts') {
     return {plugins: [...plugins, 'typescript']};
   }
-  const jsOptions: ParserOptions = {plugins: [...plugins, 'flow']};
+  if (fileType === 'tsx') {
+    return {plugins: [...plugins, 'typescript', 'jsx']};
+  }
+  const jsOptions: ParserOptions = {plugins: [...plugins, 'flow', 'jsx']};
   if (fileType === 'js') {
     return jsOptions;
   }
```

Comment:
just personal opinion, maybe a switch statement is less verbose and more expressive...?

Reply:
I stuck with conditionals to be consistent with the other methods. However, on looking at it a bit more, the `getFileType` function isn't exposed, making the pair of conditionals a bit redundant. How about this version that simply collapses them?